client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -115,9 +120,12 @@ function StockDefineLotsModalController(
    * if the fast insert option is enable do this :
    * - add new row automatically on blur
    * - set the focus in the new row
-   * @param {string} rowLot the row.entity.lot string
+   * @param {string} rowLot the row.entity.lot string/object
    */
   function onLotBlur(rowLot) {
+    // NOTE: rowLot will be an object if an existed lot was
+    //       selected from the typeahead.  Otherwise it will
+    //       be the lot name string that was typed in.
```

Comment:
I believe we need to deal with this at some place to disaggregate the object into it's component parts.  Where do we do that?

Reply:
I fixed this in the lots.modal.js submit() function.   In the past we only dealt with 'lot' as a string.  But now since we have the drop-down, 'lot' is actually an object during the typeahead.  I fixed the submit function to do the necessary conversion if an existing lot was selected.  I've tested it and it seems to work.